src/main/java/it/gov/pagopa/pu/sil/exception/DebtPositionsResponseErrorHandler.java

```diff
@@ -0,0 +1,106 @@
+package it.gov.pagopa.pu.sil.exception;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.gov.pagopa.pu.debtpositions.dto.generated.DebtPositionErrorDTO;
+import it.gov.pagopa.pu.sil.config.rest.RestTemplateConfig;
+import it.gov.pagopa.pu.sil.connector.debtpositions.config.DebtPositionsApiClientConfig;
+import it.gov.pagopa.pu.sil.enums.SilFaults;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.ResponseErrorHandler;
+
+@Slf4j
+public class DebtPositionsResponseErrorHandler extends
+  DefaultResponseErrorHandler {
+
+  private static final String DEBT_POSITIONS = "DEBT-POSITIONS";
+
+  private final ObjectMapper objectMapper;
+  private final ResponseErrorHandler errorLoggerHandler;
+
+  public DebtPositionsResponseErrorHandler(DebtPositionsApiClientConfig clientConfig,
+    ObjectMapper objectMapper) {
+    this.errorLoggerHandler = clientConfig.isPrintBodyWhenError() ?
+      RestTemplateConfig.bodyPrinterWhenError(DEBT_POSITIONS)
+      : null;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public void handleError(ClientHttpResponse response,
+    HttpStatusCode statusCode, URI url, HttpMethod method) throws IOException {
+    try {
+      if (errorLoggerHandler != null) {
+        errorLoggerHandler.handleError(url, method, response);
+      } else {
+        super.handleError(response, statusCode, url, method);
+      }
+    } catch (HttpStatusCodeException exception) {
+      if (statusCode.is4xxClientError()) {
+        String responseBody = exception.getResponseBodyAsString();
+        DebtPositionErrorDTO debtPositionErrorDTO;
+
+        try {
+          debtPositionErrorDTO = objectMapper.readValue(responseBody, DebtPositionErrorDTO.class);
+        } catch (IOException ex) {
+          log.error("Cannot deserialize DebtPositionError message from body: {}", responseBody, ex);
+          throw new SilFaultException(SilFaults.PAA_SYSTEM_ERROR, "Failed to parse DebtPositionErrorDTO from HTTP body: " + ex);
```

Comment:
this is wrong... the fact that we are not able to read the original error should no change the error to return
```suggestion
          log.error("Cannot deserialize DebtPositionError message from body: {}", responseBody, ex);
          throw ex;
```